lib/ansible/executor/module_common.py

```diff
@@ -1334,13 +1350,21 @@ def get_action_args_with_defaults(action, args, defaults, templar):
         module_defaults = templar.template(module_defaults)
 
         # deal with configured group defaults first
-        if action in C.config.module_defaults_groups:
-            for group in C.config.module_defaults_groups.get(action, []):
-                tmp_args.update((module_defaults.get('group/{0}'.format(group)) or {}).copy())
+        for group_name in collection_groups:
```

Comment:
You should probably turn this around- pick up all the `group/{thing}` prefixed module defaults, then chain lookup `{thing}` -> group -> collection list. It both cuts down on the amount of stuff you're having to resolve, and is more compatible with the v.next version of the feature where we're directly querying things from collections (otherwise we'd have to pre-scan all collections for their action_groups, which is a Bad Thing). It would also reduce the need for intermediate lookups of the group in question when it's time to apply the defaults themselves, since you'd be looping over the defaults you're looking to apply, if that makes sense...

Reply:
Done